README.md

```diff
@@ -17,6 +17,43 @@ For Java 7 use a version <= 0.7.1
 
 For Play 2.7, it requires `play.allowGlobalApplication = true` in application.conf.
 
+## Usage
+
+The library comprises two central classes, `CacheMongoRepository`, and `CacheController`.
+
+Most use-cases should use the `CacheMongoRepository` unless you have a concrete need to expose your cache via HTTP. The Controller class wraps the Repository class in a Play Controller, and can be wired directly into a `routes` file.
+
+Example usage of `CacheMongoRepository` in Scala:
+
+```scala
+@Singleton
+class SessionCacheRepository @Inject()(
+  @Named("mongodb.session.expireAfterSeconds") expireAfterSeconds: Int,
+  mongo: ReactiveMongoComponent)(implicit ec: ExecutionContext)
+    extends CacheMongoRepository("sessions", expireAfterSeconds)(mongo.mongoConnector.db, ec)
+```
+
+This creates a repository class, which can be called from elsewhere in your application. You should only create this class on application start-up, rather than dynamically calling `new`, as the class constructor will check for, and optionally create, database indexes.
+
+The `CacheMongoRepository` class exposes a method `createOrUpdate` that is used to upsert data into the cache.
+
+```scala
+def createOrUpdate(id: Id, key: String, toCache: JsValue): Future[DatabaseUpdate[Cache]]
+```
+
+The JSON structure that is cached in Mongo:
+
+```json
+{
+	"id": {
+		"key1": "value1",
+		"key2": "value1"
+	}
+}
+```
+
+This structure allows caching multiple keys against an ID. Simple use-cases can hardcode `key` to a constant, and provide a cache of ID to value.
```

Comment:
is it worth just pointing out that the reason for this structure is to ensure that all keys under the same id will have the same lifespan (like a session)?